src/mc_control/fsm/Executor.cpp

```diff
@@ -205,7 +205,11 @@ void Executor::next(Controller & ctl)
     }
     for(const auto & s : transition_map_.transitions(next_state_))
     {
-      gui->addElement(category_, mc_rtc::gui::Button("Force transition to " + s, [this, s]() { this->resume(s); }));
+      std::string buttonText = "Force transition to " + s;
+      if(!gui->hasElement(category_, buttonText))
+      {
+        gui->addElement(category_, mc_rtc::gui::Button(buttonText, [this, s]() { this->resume(s); }));
+      }
```

Comment:
It can actually happen very easily:

```yaml
transition:
- [A, Output, B, Strict]
- [A, Other Output, B, Strict]
```

Now state `A` has two transitions to the same state `B` (valid), which tries to generate two "Force transition" buttons with name `Force transition to B`, and wrongly shows an error to the user. This example is a perfectly valid use-case, and I personally tend to abuse the transition map to generate automatic choices for the user based on the forced transitions. This is not necessarily the cleanest way of doing it, but it's pretty convenient.

This is the only case I can see that would lead to duplicated force transition buttons, and it happens in a legitimate case, so I think it's fine like this. It does deserve a comment in the code though!

Reply:
Actually, this might be better:

```cpp
std::unordered_set<std::string> TransitionMap::transitions(const std::string & state) const
{                                                                             
  std::unordered_set<std::string> ret;                                       
  for(const auto & t : map_)                                               
  {                                                                                 
    if(t.first.first == state)                                    
    {                                                                         
      ret.insert(t.second.state);                                                   
    }                                                                              
  }                                                                            
  return ret;                                                                  
}  
```

Reply:
Oh, good catch I hadn't thought of that :)